spec/Data_Parallelism.tex

```diff
@@ -169,14 +169,25 @@ \subsection{Syntax}
 The handling of the outer variables within the forall expression and
 the role of \sntx{task-intent-clause} are defined in \rsec{Forall_Intents}.
 
-\subsection{Execution and Serializability}
-\label{Forall_Expression_Execution_and_Serializability}
+\subsection{Execution}
+\label{Forall_Expression_Execution}
 \index{expressions!forall!semantics}
 
-The forall expression executes a forall loop (\rsec{Forall}),
-evaluates the body expression on each iteration of the loop, and
-returns the resulting values as a collection.  The size and shape of
-that collection are determined by the iteratable-expression.
+The forall expression is an iterator that executes a forall loop (\rsec{Forall}),
+evaluates the body expression on each iteration of the loop,
+and yields each resulting value.
+
+When a forall expression is used to initialize a variable, such as
+%\begin{chapel}
+%var X = forall /*iteratable-expression*/ do /*expression*/;
+%\end{chapel}
+\\ %formatting
+{\tt \mbox{$$ $$ $$ $$ $$} {\bf var} X = {\bf forall} /*iteratable-expression*/ {\bf do} /*expression*/;}
+\\ %formatting
```

Comment:
I'd probably just make this a complete example, such as
```
\begin{chapel}
var A = forall iterableExpression() do valueToYield();
\end{chapel}
```

the formatting here seems finnicky and unnecessary.